static/app/views/performance/traceDetails/traceViewDetailPanel.tsx

```diff
@@ -0,0 +1,504 @@
+import {generateProfileFlamechartRoute} from 'sentry/utils/profiling/routes';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+import {isCustomMeasurement} from 'sentry/views/dashboards/utils';
+import DetailPanel from 'sentry/views/starfish/components/detailPanel';
+
+import {transactionSummaryRouteWithQuery} from '../transactionSummary/utils';
+
+import {EventDetail} from './newTraceDetailsContent';
+import {Row, Tags} from './styles';
+
+type DetailPanelProps = {
+  detail: EventDetail | undefined;
+  onClose: () => void;
+};
+
+type EventDetailProps = {
+  detail: EventDetail;
+  location: Location;
+  organization: Organization;
+};
+
+function OpsBreakdown({event}: {event: EventTransaction}) {
+  const [showingAll, setShowingAll] = useState(false);
+  const breakdown = event && generateStats(event, {type: 'no_filter'});
+
+  if (!breakdown) {
+    return null;
+  }
+
+  const renderText = showingAll ? t('Show less') : t('Show more') + '...';
+  return (
+    breakdown && (
+      <Row
+        title={
+          <FlexBox style={{gap: '5px'}}>
+            Ops Breakdown
+            <QuestionTooltip
+              title={t('Applicable to the children of this event only')}
+              size="xs"
+            />
+          </FlexBox>
+        }
+      >
+        <div style={{display: 'flex', flexDirection: 'column', gap: space(0.25)}}>
+          {breakdown.slice(0, showingAll ? breakdown.length : 5).map((currOp, index) => {
+            const {name, percentage, totalInterval} = currOp;
+
+            const operationName = typeof name === 'string' ? name : t('Other');
+            const durLabel = Math.round(totalInterval * 1000 * 100) / 100;
+            const pctLabel = isFinite(percentage) ? Math.round(percentage * 100) : '∞';
+
+            return (
+              <div key={index}>
+                {operationName}: {durLabel}ms ({pctLabel}%)
+              </div>
+            );
+          })}
+          {breakdown.length > 5 && (
+            <a onClick={() => setShowingAll(prev => !prev)}>{renderText}</a>
+          )}
+        </div>
+      </Row>
+    )
+  );
+}
+
+function BreadCrumbsSection({
+  event,
+  organization,
+}: {
+  event: EventTransaction;
+  organization: Organization;
+}) {
+  const [showBreadCrumbs, setShowBreadCrumbs] = useState(false);
+  const breadCrumbsContainerRef = createRef<HTMLDivElement>();
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (showBreadCrumbs) {
+        breadCrumbsContainerRef.current?.scrollIntoView({
+          behavior: 'smooth',
+          block: 'end',
+        });
+      }
+    }, 100);
+  }, [showBreadCrumbs, breadCrumbsContainerRef]);
+
+  const matchingEntry: EntryBreadcrumbs | undefined = event?.entries.find(
+    entry => entry.type === EntryType.BREADCRUMBS
+  ) as EntryBreadcrumbs;
```

Comment:
```suggestion
  const matchingEntry: EntryBreadcrumbs | undefined = event?.entries.find(
    (entry): entry is EntryBreadcrumbs => entry.type === EntryType.BREADCRUMBS
  );
```

It's a good idea to avoid `as X` whenever possible. In this case `.find()` could return `undefined` which you're properly checking for below, but because of forcing the type with `as EntryBreadcrumbs`, typescript actually doesn't recognize that it's possibly undefined. i.e. if you removed the check on line 143, it won't complain when you try to access values on line 167 since you've overwritten the type

Adding this annotation to functions tells typescript in downstream code to maintain the type checking we just did through `.find()`

Reply:
Makes sense to me thanks